internal/configs/test_file.go

```diff
@@ -396,10 +397,17 @@ func loadTestFile(body hcl.Body) (*TestFile, hcl.Diagnostics) {
 			}
 
 		case "provider":
-			provider, providerDiags := decodeProviderBlock(block)
+			providerBlock, providerDiags := decodeProviderBlock(block)
 			diags = append(diags, providerDiags...)
-			if provider != nil {
-				tf.Providers[provider.moduleUniqueKey()] = provider
+			if providerBlock != nil {
+				providers, diagsStatic := providerBlock.decodeStaticFields(nil)
+				diags = append(diags, diagsStatic...)
+				if diagsStatic.HasErrors() {
+					continue
+				}
+				// providers only ever has 1 element for nul static fields.
+				provider := providers[0]
```

Comment:
Even though the comment is currently true, let's do a for-loop instead just to be safe against future enhancements